app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -230,6 +236,11 @@ class HomeFragment : Fragment() {
             ::dispatchModeChanges
         )
 
+        val blocklistProvider = LocalHomescreenBlocklistProvider(
+            lifecycleScope,
+            requireContext().homescreenBlocklistDataStore
+        )
```

Comment:
I think we tend to init dependencies lazily on [Components](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/components/Components.kt) class :)